modules/vstudio/_preload.lua

```diff
@@ -21,6 +21,19 @@
 	include("vs2015.lua")
 	include("vs2017.lua")
 
+
+	p.api.register {
+		name = "debuggerflavor",
+		scope = "config",
+		kind = "string",
+		allowed = {
+			"WindowsLocalDebugger",
+			"WindowsRemoteDebugger",
+			"WebBrowserDebugger",
+			"WebServiceDebugger"
```

Comment:
These could simply be:
```lua
"Local",
"Remote",
"WebBrowser",
"WebService",
```
I would recommend not using the VS element values for the API values. VS changes the values sometimes and you'll end up needing a lookup table anyway, others adding support for VS extensions might need to use different values too.